osquery/database/database.cpp

```diff
@@ -674,14 +674,15 @@ Status scanDatabaseKeys(const std::string& domain,
     }
     return status;
   } else if (!DatabasePlugin::kDBInitialized) {
-    throw std::runtime_error("Cannot get database values");
+    throw std::runtime_error("Cannot scan database keys");
   } else {
     auto plugin = getDatabasePlugin();
     return plugin->scan(domain, keys, prefix, max);
   }
 }
 
 void resetDatabase() {
+  WriteLock lock(kDatabaseReset);
```

Comment:
Can you elaborate please? I thought Registry::getActive is a simple map lookup. Although setActive seems to acquire lock! Either way, if API calls reset, this change will not work.